src/GovukNotify/Client/HttpClientWrapper.cs

```diff
@@ -28,12 +32,29 @@ public HttpClientWrapper(HttpClient client)
 
         public void Dispose()
         {
-            _client.Dispose();
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            if (disposing)
+            {
+                _client.Dispose();
+            }
+
+            _disposed = true;
         }
```

Comment:
This change is to implement the dispose pattern as recommended by Microsoft:

https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose
https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/dispose-pattern